src/theory/quantifiers/sygus_grammar_norm.cpp

```diff
@@ -118,7 +118,8 @@ void SygusGrammarNorm::TypeObject::buildDatatype(SygusGrammarNorm* sygus_norm,
   Trace("sygus-grammar-normalize") << "---------------------------------\n";
 }
 
-std::map<TypeNode, std::vector<Kind>> SygusGrammarNorm::TransfChain::d_chain_ops = {};
+std::map<TypeNode, std::vector<Kind>>
```

Comment:
As I mentioned, I would not create these maps.

If you want to cache the results of isChainable/isId, you should use a map that matches the input to those functions.

Reply:
I got rid of them. What do you think it would be the best way to cache the result of ``TermUtil::mkTypeValue(tn, 0)``?